lib/components/TimePicker.js

```diff
@@ -41,21 +41,60 @@ const TimePicker = forwardRef(
       error = "",
       defaultValue,
       value,
+      customTarget,
       ...otherProps
     },
     ref
   ) => {
     const id = useId(otherProps.id);
     const timePickerRef = useSyncedRef(ref);
-
+    const [currentValue, setCurrentValue] = useState(defaultValue);
+    const [inputHide, setInputHide] = useState(false);
     const Component = timePickerTypes[type.toLowerCase()];
 
     const errorId = `error_${id}`;
-
     return (
       <div className="neeto-ui-input__wrapper">
         {label && <Label>{label}</Label>}
-        <Component
+        {customTarget ? (
+          <div className="relative">
+            <div>
+
+              {typeof customTarget === "function" ? customTarget(currentValue) : customTarget}
+            </div>
+            <div className={
+              classnames("neeto-ui-custom-target-date-input",
+                { "neeto-ui-custom-target-date-input--hidden": inputHide })} >
+              <Component
+                ref={timePickerRef}
+                format={format}
+                hourStep={interval.hourStep}
+                minuteStep={interval.minuteStep}
+                onChange={(dateObj) => {
+                  setCurrentValue(dateObj);
+                  onChange();
+                }}
+                className="h-full w-full"
+                secondStep={interval.secondStep}
+                disabled={disabled}
+                onOpenChange={() => {
+                  setTimeout(() => {
+                    setInputHide(!inputHide);
+                  }, 100);
+                }}
+                allowClear={false}
+                onCalendarChange={(dateObj) => {
+                  setCurrentValue(dateObj);
+                }}
+                {...otherProps}
+                picker="time"
+                mode={undefined}
+                defaultValue={convertToDayjsObjects(defaultValue)}
+                value={convertToDayjsObjects(value)}
+              />
+            </div>
+          </div>
+        ) : <Component
```

Comment:
Can we extract the logic to be a wrapper component that conditionally adds the `customTarget` if defined and reuse the same in `DatePicker` as well.